pkg/util/slices/sorted.go

```diff
@@ -124,3 +118,9 @@ func (s *Sorted[T]) Copy() *Sorted[T] {
 		cmp:   s.cmp,
 	}
 }
+
+func (s *Sorted[T]) find(item T) (int, bool) {
+	return sort.Find(s.Len(), func(i int) int {
```

Comment:
Nitpick: `slices.BinarySearchFunc(s.items, item, s.cmp)` is a little neater, given that we're using generics.